elastic/kibana
# [Inventory rule] Fix bug in inventory rule for Inbound and Outbound traffic threshold (both preview and executor)

Fixes #177912

## Summary

This PR fixes the wrong threshold value in the preview and execution of the inventory rule type for both Inbound and Outbound traffic.

|Before|After|
|---|---|
|![image](https://github.com/elastic/kibana/assets/12370520/015761c0-31f7-4ce0-8c6c-9d889b50df97)|![image](https://github.com/elastic/kibana/assets/12370520/48434ee0-4a0d-4fee-9fd1-e60bcb31f216)|

## 🧪  How to test
- Adjust data forge package and hard code a specific value for ingress and/or egress
    - For example, I changed this [line](https://github.com/elastic/kibana/blob/main/x-pack/packages/kbn-data-forge/src/data_sources/fake_hosts/index.ts#L100) to `bytes: 64,`, then when I created a rule, I always had 8.5 bit/s (64 * 8 = 512, and 512 / 60 = 8.5) Before the fix, if I added 8 bit/s as threshold, I didn't get any alert because the value that was compared to 1.7 bit/s (8.5 / 8) was 8 instead of 1 (8 / 8) which was incorrect.
    <img src="https://github.com/elastic/kibana/assets/12370520/a8c372c5-f5f7-4168-b329-168dbfd20101" width=500 />

- Run the data forge command to generate some data
```
node x-pack/scripts/data_forge.js \
  --events-per-cycle 7 \
  --lookback now-20m \
  --install-kibana-assets \
  --dataset fake_hosts
```
- Create an inventory rule for egress/ingress depending on which one you adjusted, and make sure the rule generates an alert correctly and the data shown in the alert table and flyout are also correct.
